documentation/src/docs/security/securing-distributed-mode-cluster.md

```diff
@@ -0,0 +1,311 @@
+    -storepass changeit
+
+   # Optionally, verify the contents of the generated file:
+   $ keytool -list -v -keystore controller01.jks -storepass changeit
+   ```
+
+2. Generate CSR for each service. It helps to think of a CSR as an application for getting a certificate signed by a trusted authority.
+
+   A CSR is typically generated on the same server on which the service is planned to be installed. Some organizations generate these outside of the servers in a central location and then distribute the resulting certificates to the services for use.
+
+   ```
+   $ keytool -keystore controller01.jks -alias controller01 -certreq -file controller01.csr \
+       -storepass changeit
+
+   # Optionally, inspect the contents of the CSR file. The CSR is created in Base-64 encoded PEM format.
+   $ openssl req -in controller01.csr -noout -text
+   ```
+
+3. Submit the CSR to a CA and obtain a signed certificate for each service.
+
+   If you are using a public or internal CA service, follow that CA's process for submitting the CSR and obtaining a signed certificate. To use the custom CA generated using the steps mentioned earlier or an internal CA certificate/key bundle, use the following command to generate a CA-signed server certificate in PEM format:
+
+   ```
+   $ openssl x509 -req -CA ca-cert -CAkey ca-key -in controller01.csr -out controller01.pem \
+        -days 3650 -CAcreateserial -passin pass:changeit
+   ```
+
+4. Prepare a keystore containing the signed server certificate and the CA's certificate.
+
+   ```
+   # Import the CA certificate into a new keystore file.
+   $ keytool -keystore controller01.server.jks -alias CARoot -noprompt \
+           -import -file ca-cert -storepass changeit
+
+   # Import the signed server certificate into the keystore.
+   $ keytool -keystore controller01.server.jks -alias controller01 -noprompt \
+           -import -file controller01.pem -storepass changeit
+   ```
+
+5. Export each server's key into a separate .pem file.
+
+   This is a two step process. First, convert the server's keystore in `.jks` format into `.p12` format.
+
+   ```
+   keytool -importkeystore -srckeystore controller01.jks  \
+                           -destkeystore controller01.p12 \
+                           -srcstoretype jks -deststoretype pkcs12 \
+                           -srcstorepass changeit -deststorepass changeit
+   ```
+
+   Then, export the private key of the server into a `.pem` file. Note that the generated .pem file is not protected by a password, even though the key itself is password-protected, as we are using the `-nodes` flag. So, be sure to protect it using operating system's technical controls as well as procedural controls.
+
+   ```
+   openssl pkcs12 -in controller01.p12 -out controller01.key.pem -passin pass:1111_aaaa -nodes
+   ```
+
+Step 5 concludes this stage, and you are now all set to deploy the certificates and other material in Pravega.
```

Comment:
Step 5 concludes this stage, and  now  deploying the certificates and other material in Pravega can be implemented.

Reply:
Done.